src/app/code/community/Afterpay/Afterpay/Helper/Data.php

```diff
@@ -401,6 +401,36 @@ public function giftCardsPlaceOrder()
 
         return true;
     }
-    
+
+    /**
+     * Workaround to deal with Afterpay response which has been chunked and then compressed,
+     * rather than compressed and then chunked.
+     *
+     * @param Zend_Http_Response $response
+     * @return string
+     */
+    public function getChunkedBody($response){
+        try{
+            return $response->getBody();
```

Comment:
I'd rather refactor this so there's a single return statement. 
Also concern about some decode logic in the catch block.
For example, what if Zend_Http_Response::decodeGzip($body) throws exception ?

Reply:
@willtran - the original getBody can (and does) throw exceptions.  I'm ok for it to throw an exception and for that to be logged and handled further up the chain as it was previously being.  I was trying to get around a specific exception.  However also as per comments on slack from @michaelwylde , I would be more specific about which exception I'm catching.D

Noted regards the single return statement - I'll update it.

Reply:
If we know for sure that Afterpay server send the response as non-chunked and then compressed (not intermittently), I'd get rid of the first try $response->getBody() all together.
Have we contacted them about heading being wrong?

Reply:
If that was the case it is best to make it a system config that ifChunk do stuff else do other. So that if they change it at a later stage it is easy to swap over?

Reply:
@willtran  - logged as issue https://github.com/afterpay/afterpay-magento/issues/27

@michaelwylde  - The way it is would handle both cases without requiring manual intervention setting a config value?  If they fix their server, then the first "getBody" won't throw an exception and will return the correct output.